fili-core/src/main/java/com/yahoo/bard/webservice/util/StreamUtils.java

```diff
@@ -145,4 +146,137 @@
                 return castedOperand;
         };
     }
+
+    /**
+     * Build a function that can map a Map from one map to another.
+     *
+     * @param keyMapper  Function to map from the input Map's Entries to the result Map's keys
+     * @param valueMapper  Function to map from the input Map's Entries to the result Map's values
+     * @param resultSupplier  Supplier of the result map instance
+     * @param <A>  Type of the input Map's keys
+     * @param <B>  Type of the input Map's values
+     * @param <K>  Type of the result Map's keys
+     * @param <V>  Type of the result Map's values
+     * @param <R>  Type of the input Map
+     * @param <T>  Type of the result map
+     *
+     * @return a function that can map a Map from one map to another.
+     */
+    public static <A, B, K, V, R extends Map<A, B>, T extends Map<K, V>> Function<R, T> mapMap(
+            Function<? super Map.Entry<A, B>, ? extends K> keyMapper,
+            Function<? super Map.Entry<A, B>, ? extends V> valueMapper,
+            Supplier<T> resultSupplier
+    ) {
+        return (R input) -> input.entrySet().stream()
+                .collect(
+                        resultSupplier,
+                        (result, entry) -> result.put(keyMapper.apply(entry), valueMapper.apply(entry)),
+                        Map::putAll
```

Comment:
Good catch. Part of the point of this method is to smooth over the "common parts" of mapping the keys or the values of a map, like processing the entry set, and then collecting back into a map. It's not really intended for doing any reducing (that's why it's a `Function` instead of a `Collector`), but that's not really knowable from the method name or javadoc.

So, do you think it's sufficient to just add a bit to the doc indicating that this uses a "putAll" combiner (and what that means)? And do you think I need a version that allows for providing a combiner?